// @aws-amplify/ui-react-core/dist/esm/InAppMessaging/hooks/useMessage/utils.mjs
import { __rest } from '../../../node_modules/tslib/tslib.es6.mjs';
import { ConsoleLogger } from '@aws-amplify/core';

const logger = new ConsoleLogger('Notifications.InAppMessaging');
const positions = {
    BOTTOM_BANNER: 'bottom',
    MIDDLE_BANNER: 'middle',
    TOP_BANNER: 'top',
};
const getPositionProp = (layout) => positions[layout];
const getActionHandler = (actionParams, onMessageAction, onActionCallback) => ({
    onAction() {
        try {
            onMessageAction(actionParams);
        }
        catch (e) {
            logger.error(`Message action failure: ${e}`);
        }
        finally {
            onActionCallback();
        }
    },
});
const getButtonProps = (_a, onMessageAction, onActionCallback) => {
    var { action, url } = _a, baseButtonProps = __rest(_a, ["action", "url"]);
    return (Object.assign(Object.assign({}, baseButtonProps), getActionHandler({ action, url }, onMessageAction, onActionCallback)));
};
const getContentProps = (content, onMessageAction, onActionCallback) => {
    const props = {};
    if (!content) {
        return props;
    }
    const { primaryButton, secondaryButton } = content, restContent = __rest(content, ["primaryButton", "secondaryButton"]);
    if (primaryButton) {
        props.primaryButton = getButtonProps(primaryButton, onMessageAction, onActionCallback);
    }
    if (secondaryButton) {
        props.secondaryButton = getButtonProps(secondaryButton, onMessageAction, onActionCallback);
    }
    return Object.assign(Object.assign({}, props), restContent);
};

export { getActionHandler, getContentProps, getPositionProp };
